hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,303 @@
+
+        # remember if we collect any alembic/mantra nodes
+        self._alembic_nodes_collected = False
+        self._mantra_nodes_collected = False
+
+        # methods to collect tk alembic/mantra nodes if the app is installed
+        self.collect_tk_alembicnodes(item)
+        self.collect_tk_mantranodes(item)
+
+        # collect other, non-toolkit outputs to present for publishing
+        self.collect_node_outputs(item)
+
+    def collect_current_houdini_session(self, settings, parent_item):
+        """
+        Creates an item that represents the current houdini session.
+
+        :param dict settings: Configured settings for this collector
+        :param parent_item: Parent Item instance
+
+        :returns: Item of type houdini.session
+        """
+
+        publisher = self.parent
+
+        # get the path to the current file
+        path = hou.hipFile.path()
+
+        # determine the display name for the item
+        if path:
+            file_info = publisher.util.get_file_path_components(path)
+            display_name = file_info["filename"]
+        else:
+            display_name = "Current Houdini Session"
+
+        # create the session item for the publish hierarchy
+        session_item = parent_item.create_item(
+            "houdini.session",
+            "Houdini File",
+            display_name
+        )
+
+        # get the icon path to display for this item
+        icon_path = os.path.join(
+            self.disk_location,
+            os.pardir,
+            "icons",
+            "houdini.png"
+        )
+        session_item.set_icon_from_path(icon_path)
+
+        # if a work file template is defined, add it to the item properties so
+        # that it can be used by attached publish plugins
+        work_template_setting = settings.get("Work file Template")
+        if work_template_setting:
+            work_template = publisher.engine.get_template_by_name(
+                work_template_setting.value)
+
+            # store the template on the item for use by publish plugins. we
+            # can't evaluate the fields here because there's no guarantee the
+            # current session path won't change once the item has been created.
+            # the attached publish plugins will need to resolve the fields at
+            # execution time.
+            session_item.properties["work_file_template"] = work_template
+            self.logger.debug(
+                "Work file template defined for Houdini collection.")
+
+        self.logger.info("Collected current Houdini session")
+
+        return session_item
+
+    def collect_node_outputs(self, parent_item):
+        """
+        Creates items for known output nodes
+
+        :param parent_item: Parent Item instance
+        """
+
+        for node_category in _HOUDINI_OUTPUTS:
+            for node_type in _HOUDINI_OUTPUTS[node_category]:
+
+                if node_type == "alembic" and self._alembic_nodes_collected:
```

Comment:
Added these two conditionals. If the tk alembic/mantra nodes were collected, don't collect regular outputs. This avoid attempting to publish a tk node that contains a regular houdini node. I think it also makes sense that if the tk nodes are in play, publishing regular mantra/alembic nodes may be discouraged? Curious what you think.